### PR TITLE
Add typings for @lerna/package & @lerna/project

### PR DESCRIPTION
**Pull request by donmahallem**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/lerna__package/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Type definitions for @lerna/package 4.0
+// Project: https://github.com/lerna/lerna/tree/main/core/package (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/lerna__package/index.d.ts:2]:
```suggestion
// Project: https://github.com/lerna/lerna/tree/main/core/package
```

File: types/lerna__package/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+} & { [key: string]: any; };
+/**
+ * Lerna's internal representation of a local package, with
+ * many values resolved directly from the original JSON.
+ */
+export class Package {
+    /**
+     * Create a Package instance from parameters, possibly reusing existing instance.
+     * @param ref A path to a package.json file, Package instance, or JSON object
+     * @param If `ref` is a JSON object, this is the location of the manifest
```
Comment (peterblazejewicz) [inline types/lerna__package/index.d.ts:32]:
```suggestion
     * @param [dir] If `ref` is a JSON object, this is the location of the manifest
```

File: types/lerna__project/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// Type definitions for @lerna/project 4.0
+// Project: https://github.com/lerna/lerna/tree/main/core/project (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/lerna__project/index.d.ts:2]:
```suggestion
// Project: https://github.com/lerna/lerna/tree/main/core/project
```

File: types/lerna__project/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+ * A representation of the entire project managed by Lerna.
+ *
+ * Wherever the lerna.json file is located, that is the project root.
+ * All package globs are rooted from this location.
+ */
+export class Project {
+    /**
+     * @param cwd defaults to process.cwd()
+     */
+    static getPackges(cwd?: string): Promise<Package[]>;
```
Comment (peterblazejewicz) [inline types/lerna__project/index.d.ts:27]:
```suggestion
    static getPackages(cwd?: string): Promise<Package[]>;
```

File: types/lerna__project/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    /**
+     * @param cwd defaults to process.cwd()
+     */
+    static getPackgesSync(cwd?: string): Package[];
```
Comment (peterblazejewicz) [inline types/lerna__project/index.d.ts:31]:
```suggestion
    static getPackagesSync(cwd?: string): Package[];
```

File: types/lerna__project/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    getPackages(): Promise<Package[]>;
+    getPackagesSync(): Package[];
+    getPackageLicensePaths(): string[];
+    isIndependent(): boolean;
+    rootPath: string;
+    rootConfigLocation: string;
+    config: ProjectConfig;
+}
+export function getPackages(cwd?: string): Promise<Package[]>;
+export function getPackagesSync(cwd?: string): Package[];
```
Comment (peterblazejewicz) [inline types/lerna__project/index.d.ts:49]:
```suggestion
export const getPackages: typeof Project.getPackages;
export const getPackagesSync: typeof Project.getPackagesSync;
```

Comment (donmahallem) [reply types/lerna__project/index.d.ts:49]: Thanks. Was trying to infer the types the same way but didn't get it to work.

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@donmahallem thx!

**Comment by donmahallem** [comment]
Ready to merge

**Pull request merged**